Activate address sanitizer in CI, on Linux and Windows (#2219)
Azure/azure-sdk-for-c: README.md

<td>This option can be set to any of the next values:<br>- No_value: default value is used and no_platform library is used.<br>- "POSIX": Provides implementation for Linux and Mac systems.<br>- "WIN32": Provides platform implementation for Windows based system<br>- "CUSTOM": Tells cmake to use an specific implementation provided by user. When setting this option, user must provide an implementation library and set option `AZ_CUSTOM_PLATFORM_IMPL_NAME` with the name of the library (i.e. <code>-DAZ_PLATFORM_IMPL=CUSTOM -DAZ_CUSTOM_PLATFORM_IMPL_NAME=user_platform_lib</code>). cmake will look for this library to link az_core</td>
<td>No_value</td>
</tr>
<tr>
<td>ADDRESS_SANITIZER</td>

> <td>This option enables asan (address sanitizer). This should work on at least Windows and Linux and will catch memory errors at runtime. Do not use this option in production

Reviewer: Is it relevant to mention it will "at least" work on Windows/Linux? What do you mean by that?
nit: I would remove the word "should".

```suggestion
<td>This option enables asan (address sanitizer). This works on Windows and Linux to catch memory errors at runtime. Do not use this option in production
```

Author: It might work on MacOS, or other platforms that support asan. But I haven't tried it.

Author: if you set the option on those other platforms it will still try and add the compiler and linker arguments, so if they use a "normal" gnu style toolchain it may well work fine.

> as asan introduces trivial code execution vectors via environment variables.</td>

Reviewer: What do you mean by the following?
> asan introduces trivial code execution vectors via environment variables

Also, did you mean non-trivial, instead of trivial?

Author: no, they are trivial to exploit. It's not just that but asan leaks information about the binary and can be used to easily defeat ASLR. Malicious memory errors are also not a target of asan, so it's pretty easy to bypass if you are actually trying to do so.

> as asan introduces trivial code execution vectors via environment variables.</td>

Reviewer: consistency nit: don't have explicit line split/wrap for the description, and keep it on the same line
